### PR TITLE
storage service edit

### PR DESCRIPTION
**Pull request by OrGur1987**
belongs to issue: https://github.com/ManageIQ/manageiq-ui-classic/issues/8728

depends upon:
- [x] manageiq/manageiq#22425

adds editing/update option to storage service controller and app yaml

### CODE REVIEW

**Comment by OrGur1987** [comment]
@bdunne hi, this is ready for review
thanks

File: config/api.yml [hunk 1]
```diff
@@ -4231,6 +4231,8 @@
       :post:
       - :name: create
         :identifier: storage_service_new
+      - :name: edit
+        :identifier: physical_storage_edit
```
Comment (kbrock) [inline config/api.yml:4235]: did you want `storage_service_edit` here?

**Review by kbrock — CHANGES_REQUESTED** [review]
this looks great. just that one change

**Comment by kbrock** [comment]
Also, is there a way to add a spec here?

**Comment by OrGur1987** [comment]
> Also, is there a way to add a spec here?

should I use `spec/models/cloud_volume_spec.rb` as an example?
the specs there test queuing methods

**Comment by OrGur1987** [comment]
> > Also, is there a way to add a spec here?
> 
> should I use `spec/models/cloud_volume_spec.rb` as an example? the specs there test queuing methods

@kbrock

**Comment by kbrock** [comment]
@OrGur1987 good point. you can't test that the records have been modified because this is just a request for the provider.

Yea, those tests are as good as any.

**Comment by OrGur1987** [comment]
> @OrGur1987 good point. you can't test that the records have been modified because this is just a request for the provider.
> 
> Yes, base your tests upon those.

added an edit spec to `spec/requests/storage_services_spec.rb`.

File: spec/requests/storage_services_spec.rb [hunk 2]
```diff
@@ -79,4 +79,19 @@
 
     expect(response).to have_http_status(:ok)
   end
+
+  context 'Storage Services edit action' do
+    it "PUT /api/storage_services/:id'" do
+      require 'byebug'
+      byebug
```
Comment (kbrock) [inline spec/requests/storage_services_spec.rb:86]: oops. how did this get in there? ;)

File: spec/requests/storage_services_spec.rb [hunk 3]
```diff
@@ -79,4 +79,19 @@
...
+      provider = FactoryBot.create(:ems_autosde, :name => 'Autosde')
+      storage_service = FactoryBot.create("ManageIQ::Providers::Autosde::StorageManager::StorageService", :name => 'test_service1', :ext_management_system => provider)
```
Comment (kbrock) [inline spec/requests/storage_services_spec.rb:88]:
```suggestion
      storage_service = FactoryBot.create(:autosde_storage_manager, :ext_management_system => provider)
```

**Comment by kbrock** [comment]
@OrGur1987 This looks great. Just one or two tweaks and we can get it in right away.

Thanks for doing this

File: spec/requests/storage_services_spec.rb (hunk 3 above)
Comment (OrGur1987) [reply spec/requests/storage_services_spec.rb:85]: extracted `provider` to `let()`.
extracting `storage_service` didn't quite work out with the `delete` and `get` specs.. 

in the `edit` spec, for some reason when I use `FactoryBot.create(:storage_service, ... )` the service does not support `update`, but it does when I use `FactoryBot.create("ManageIQ::Providers::Autosde::StorageManager::StorageService",...)`.

do u have any idea why? maybe it references the core repo `storage_service` model and not the provider's one?
it supposedly doesn't support `create` and `delete` as well, but those spec do pass..

```
    83:   context 'Storage Services edit action' do
    84:     it "PUT /api/storage_services/:id'" do
    85:       storage_service1 = FactoryBot.create("ManageIQ::Providers::Autosde::StorageManager::StorageService", :name => 'test_service1', :ext_management_system => provider)
    86:       storage_service2 = FactoryBot.create(:storage_service, :name => 'test_service1', :ext_management_system => provider)
    87: 
    88:       require 'byebug'
    89:       byebug
    90: 
=>  91:       api_basic_authorize('storage_service_edit')
    92:       put(api_storage_service_url(nil, storage_service))
    93: 
    94: 
    95:       expect(response.parsed_body["message"]).to include("Updating")
    96:       expect(response).to have_http_status(:ok)
    97:     end
    98:   end
    99: 
   100: end
(byebug) storage_service1.supports_update?
true
(byebug) storage_service2.supports_update?
false
(byebug) storage_service2.supports_create?
false
(byebug) storage_service2.supports_delete?
false
(byebug) storage_service1.supports_delete?
true

```

**Comment by OrGur1987** [comment]
> @OrGur1987 This looks great. Just one or two tweaks and we can get it in right away.
> 
> Thanks for doing this

@kbrock is spec ok like that?

**Comment by OrGur1987** [comment]
@kbrock  
kind reminder :)

File: spec/requests/storage_services_spec.rb (hunk 3 above)
Comment (kbrock) [reply spec/requests/storage_services_spec.rb:85]: sorry.

We tend to use `stub_supports` to ensure that you get the expected `supports?`.
In the far far future, we would like to change all these tests to use a generic providers with a bunch of expectations on them.

**Comment by kbrock** [comment]
@OrGur1987 thanks for the ping. I'll leave this tab open and hope it does not slip through the cracks again.

Could you see my notes around the class of the `storage_service2`?
Also, don't set any attributes (i.e.: name) in the provider if it will work without it set.

**Comment by OrGur1987** [comment]
> @OrGur1987 thanks for the ping. I'll leave this tab open and hope it does not slip through the cracks again.
> 
> Could you see my notes around the class of the `storage_service2`? Also, don't set any attributes (i.e.: name) in the provider if it will work without it set.
> 
> Not sure why the factory is a sticking point for me

@kbrock 
`stub_supports` fixed it thnx :)
removed the name from the provider

**Pull request merged**